### PR TITLE
54 add delete needs

### PR DESCRIPTION
**Pull request by rik86future**

### CODE REVIEW

**Comment by rik86future** [comment]
["Delete" and "Edit" Support actions, Call Logs and Notes created in error- Contact Agents and Managers](https://trello.com/c/DkPSRrjC/54-delete-and-edit-support-actions-call-logs-and-notes-created-in-error-contact-agents-and-managers)

File: app/models/need.rb [hunk 1]
```diff
@@ -80,6 +83,8 @@ def start_required_for_assessment
 
   scope :filter_by_status, ->(status) { where(status: status) }
 
+  scope :created_by, ->(user_id) { joins(:versions).where('whodunnit = ? and event = ?', user_id.to_s, 'create') }
```
Comment (jamiebuckley) [inline app/models/need.rb:86]: Wonder if we should create an index for this

File: app/policies/need_policy.rb [hunk 2]
```diff
@@ -46,6 +46,20 @@ def show?
     Pundit.policy_scope!(@user, Need).where(id: @record.id).exists?
   end
 
+  def destroy?
+    # managers can always destroy
+    return true if admin?
+
+    # check ownership of need record
+    need = Pundit.policy_scope!(@user, Need).where(id: @record.id)
```
Comment (jamiebuckley) [inline app/policies/need_policy.rb:54]: Policy scope for certain roles (MDT, Food Hub, Council Services), restricts viewable needs to those that are assigned to them, their team, or their role.

This filtering is done by the policy scope - so `Pundit.policy_scope!(@user, Need).where(id: @record.id).exists?` will return false if it is not assigned to one of those three that put it in the policy scope, even if the user created it.

Comment (rik86future) [reply app/policies/need_policy.rb:54]: ok I get it now, thanks

Comment (rik86future) [reply app/policies/need_policy.rb]: @jamiebuckley  amended now

**Comment by jamiebuckley** [comment]
Looks great

**Pull request merged**